### PR TITLE
Fixes #916

### LINKED ISSUE #916: eks-blueprints: Cannot import ouput values to create ImportClusterProvider
### Describe the bug

I have created EKS cluster using Blueprints in Stack1 and create addon's in separate stack. So, I followed all 3 proceedures mentioned in https://aws-quickstart.github.io/cdk-eks-blueprints/cluster-providers/import-cluster-provider/, but build is failing as its not able to import values from CfnOutputs

### Expected Behavior

CDK should build successfully and apply the addons to the EKS cluster.

### Current Behavior

Build fails with below error
```
<Source_Directory>/node_modules/@aws-quickstart/eks-blueprints/node_modules/aws-cdk-lib/core/lib/private/uniqueid.js:1
"use strict";Object.defineProperty(exports,"__esModule",{value:!0}),exports.makeUniqueId=void 0;const encoding_1=require("./encoding"),md5_1=require("./md5"),HIDDEN_FROM_HUMAN_ID="Resource",HIDDEN_ID="Default",PATH_SEP="/",HASH_LEN=8,MAX_HUMAN_LEN=240,MAX_ID_LEN=255;function makeUniqueId(components){if(components=components.filter(x=>x!==HIDDEN_ID),components.length===0)throw new Error("Unable to calculate a unique id for an empty set of components");const unresolvedTokens=components.filter(c=>(0,encoding_1.unresolved)(c));if(unresolvedTokens.length>0)throw new Error(`ID components may not include unresolved tokens: ${unresolvedTokens.join(",")}`);if(components.length===1){const candidate=removeNonAlphanumeric(components[0]);if(candidate.length<=MAX_ID_LEN)return candidate}const hash=pathHash(components);return removeDupes(components).filter(x=>x!==HIDDEN_FROM_HUMAN_ID).map(removeNonAlphanumeric).join("").slice(0,MAX_HUMAN_LEN)+hash}exports.makeUniqueId=makeUniqueId;function pathHash(path){return(0,md5_1.md5hash)(path.join(PATH_SEP)).slice(0,HASH_LEN).toUpperCase()}function removeNonAlphanumeric(s){return s.replace(/[^A-Za-z0-9]/g,"")}function removeDupes(path){const ret=new Array;for(const component of path)(ret.length===0||!ret[ret.length-1].endsWith(component))&&ret.push(component);return ret}
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    ^

Error: ID components may not include unresolved tokens: imported-cluster-${Token[TOKEN.974]}
    at makeUniqueId (<Source_Directory>/node_modules/@aws-quickstart/eks-blueprints/node_modules/aws-cdk-lib/core/lib/private/uniqueid.js:1:565)
    at Names.nodeUniqueId (<Source_Directory>/node_modules/@aws-quickstart/eks-blueprints/node_modules/aws-cdk-lib/core/lib/names.js:1:776)
    at KubectlProvider.getOrCreate (<Source_Directory>/node_modules/@aws-quickstart/eks-blueprints/node_modules/aws-cdk-lib/aws-eks/lib/kubectl-provider.js:1:1011)
    at new KubernetesManifest (<Source_Directory>/node_modules/@aws-quickstart/eks-blueprints/node_modules/aws-cdk-lib/aws-eks/lib/k8s-manifest.js:1:811)
    at createNamespace (<Source_Directory>/node_modules/@aws-quickstart/eks-blueprints/dist/utils/namespace-utils.js:15:12)
    at PrometheusNodeExporterAddOn.deploy (<Source_Directory>/node_modules/@aws-quickstart/eks-blueprints/dist/addons/prometheus-node-exporter/index.js:41:59)
    at new EksBlueprint (<Source_Directory>/node_modules/@aws-quickstart/eks-blueprints/dist/stacks/eks-blueprint-stack.js:197:34)
    at BlueprintBuilder.build (<Source_Directory>/node_modules/@aws-quickstart/eks-blueprints/dist/stacks/eks-blueprint-stack.js:146:16)
```

### Reproduction Steps

**Sample code:**

**Stack1**
```
const blueprint = blueprints.EksBlueprint.builder()
    .version('auto')
    .account('xxx')
    .region('xxx')
    .teams()
    .build(scope, 'simple-stack');

new CfnOutput(this, 'ClusterName', { value: blueprint.getClusterInfo().cluster.clusterName, exportName: 'ClusterName' });
    new CfnOutput(this, 'ClusterSecurityGroupId', { value: blueprint.getClusterInfo().cluster.clusterSecurityGroupId, exportName: 'ClusterSecurityGroup' });
    new CfnOutput(this, 'KubectlRoleArn', { value: blueprint.getClusterInfo().cluster.kubectlRole?.roleArn!, exportName: 'ClusterKubectlArn' });
    new CfnOutput(this, 'ClusterEndpoint', { value: blueprint.getClusterInfo().cluster.clusterEndpoint, exportName: 'ClusterEndpoint' });
    new CfnOutput(this, 'KubectlProviderHandlerRole', { value: blueprint.getClusterInfo().kubectlProvider.handlerRole.roleArn, exportName: 'KubectlProviderHandlerRoleArn' });
```

**Stack2**

```
 const prometheus_node_exporter_addon = new blueprints.addons.PrometheusNodeExporterAddOn({
      createNamespace: true,
    });

    const addOns: Array<ClusterAddOn> = [
      prometheus_node_exporter_addon,
    ];
const importClusterProvider3 = new ImportClusterProvider({
      clusterName: Fn.importValue('ClusterName'),
      version: KubernetesVersion.V1_27,
      clusterEndpoint: Fn.importValue('ClusterEndpoint'),
      openIdConnectProvider: getResource((context) =>
        new LookupOpenIdConnectProvider(
          'XXXXXX',
        ).provide(context),
      ),
      kubectlRoleArn: Fn.importValue('KubectlRoleArn'),
      clusterSecurityGroupId: Fn.importValue('ClusterSecurityGroupId'),
    });
const vpcId = 'XXXXXX';

    blueprints.EksBlueprint.builder()
      .clusterProvider(importClusterProvider3)
      .resourceProvider(blueprints.GlobalResources.Vpc, new blueprints.VpcProvider(vpcId))
      .addOns(...addOns)
      .build(scope, 'test');
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.61.0

### EKS Blueprints Version

1.13.1

### Node.js Version

18.19.0

### Environment details (OS name and version, etc.)

Mac OS 14.2.1

### Other information

_No response_

### PR DESCRIPTION
**Pull request by shapirov103**
*Issue #, if available:* #916

*Description of changes:*
Import cluster provider can handle cluster name as token. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by shapirov103** [comment]
@elamaran11  please review

File: lib/utils/id-utils.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Token } from 'aws-cdk-lib';
+import { v4 as uuid } from 'uuid';
+
+export function uniqueId() : string {
```
Comment (elamaran11) [inline lib/utils/id-utils.ts:4]: Could we add a doc on what is the usage of this utility.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
LGTM. Minor comment!

File: lib/utils/id-utils.ts (hunk 1 above)
Comment (shapirov103) [reply lib/utils/id-utils.ts:8]: of course. addressed

**Comment by shapirov103** [comment]
@elamaran11 addressed review comments

**Review by elamaran11 — APPROVED** [review]
LGTM

**Pull request merged**